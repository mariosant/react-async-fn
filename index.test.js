import React from 'react'
import testRenderer from 'react-test-renderer'
import hoc from './'

const e = React.createElement

const Component = props => <div {...props}>Hello World!</div>

const asyncFn = jest.fn()

beforeEach(jest.resetAllMocks)

test('Renders component properly', () => {
	asyncFn.mockResolvedValue('Okay')

	const Enhanced = hoc(asyncFn)(Component)
	const snapshot = testRenderer.create(<Enhanced />).toJSON()

	expect(snapshot).toMatchSnapshot()
})
