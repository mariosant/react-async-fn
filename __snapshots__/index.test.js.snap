// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component properly 1`] = `
<div
  load={[Function]}
  loading={false}
  update={[Function]}
>
  Hello World!
</div>
`;
